### ISSUE TITLE
HIDE_ATTRIBUTES doesn't hide attributes without adding a dummy attribute modifier

### ISSUE THREAD

**Issue by hapyl** [PaperSpigot/Paper#11224]
### Expected behavior

Item with tag `HIDE_ATTRIBUTES` should not show attributes:

![image](https://github.com/user-attachments/assets/0919b740-9598-4dbf-801d-affc4a251193)

### Observed/Actual behavior

Item with tag 'HIDE_ATTRIBUTES' shows attributes:

![image](https://github.com/user-attachments/assets/db94344c-fd4c-4d66-9c7d-5df0de76e3d6)

### Steps/models to reproduce

```java
final ItemStack item = new ItemStack(Material.IRON_PICKAXE);
final ItemMeta meta = item.getItemMeta();

meta.setDisplayName("Test Item");
meta.addItemFlags(ItemFlag.HIDE_ATTRIBUTES);

item.setItemMeta(meta);

player.getInventory().addItem(item);
```

### Plugin and Datapack List

```
> plugins
[11:50:28 INFO]: Server Plugins (1):
[11:50:28 INFO]: Bukkit Plugins:
[11:50:28 INFO]:  - PaperProject
> datapack list
[11:51:11 INFO]: There are 3 data pack(s) enabled: [vanilla (built-in)], [file/bukkit (world)], [paper (built-in)]
[11:51:11 INFO]: There are no more data packs available
```

### Paper version

```
> version
[11:51:37 INFO]: Checking version, please wait...
[11:51:39 INFO]: This server is running Paper version 1.21-127-master@4e6a2a1 (2024-08-07T12:50:10Z) (Implementing API version 1.21-R0.1-SNAPSHOT)
You are running the latest version
Previous version: 1.21-81-ceeb8c1 (MC: 1.21)
```

### Other

This works as intended on latest Spigot:

- If there are no modifiers and `HIDE_ATTRIBUTES` tag is present, it will add a dummy empty modifier.

Paper will only hide default attributes if there is any modifier and the tag is set, but will not add a dummy modifier:

```java
final ItemStack item = new ItemStack(Material.IRON_PICKAXE);
final ItemMeta meta = item.getItemMeta();

meta.setDisplayName("Test Item with dummy modifier");
meta.addItemFlags(ItemFlag.HIDE_ATTRIBUTES);

meta.addAttributeModifier(Attribute.GENERIC_LUCK, new AttributeModifier(
        new NamespacedKey(this, "dummy"),
        0,
        AttributeModifier.Operation.ADD_NUMBER
));

item.setItemMeta(meta);

player.getInventory().addItem(item);
```

![image](https://github.com/user-attachments/assets/c0f8506e-b551-40bf-ae1e-5ac6b5117adb)

**Comment by Malfrador** [comment]
Duplicate of https://github.com/PaperMC/Paper/issues/10655

This is working as intended.

**Comment by bluhbipo** [comment]
> Duplicate of [#10655](https://github.com/PaperMC/Paper/issues/10655)
> 
> This is working as intended. To hide something, something needs to be added first. Vanilla behavior since 1.20.5

working as of spigot 1.21.4. this is a paper specific issue

**Comment by lynxplay** [comment]
This has been resolved in 1.21.5 and onwards.
It is also to note that the spigot "fix" for this behaviour results in items with all their attributes locked, e.g. won't be updated if mojang changes the default attribute modifiers of the stacks base item.

**Comment by bluhbipo** [comment]
> This has been resolved in 1.21.5 and onwards. It is also to note that the spigot "fix" for this behaviour results in items with all their attributes locked, e.g. won't be updated if mojang changes the default attribute modifiers of the stacks base item.

well darn!
